clang-tools-extra/clang-tidy/readability/ImplicitBoolConversionCheck.cpp

```diff
@@ -39,14 +39,22 @@ AST_MATCHER(Stmt, isNULLMacroExpansion) {
 }
 
 StringRef getZeroLiteralToCompareWithForType(CastKind CastExprKind,
-                                             QualType Type,
-                                             ASTContext &Context) {
+                                             QualType Type, ASTContext &Context,
+                                             bool UseUpperCaseLiteralSuffix) {
   switch (CastExprKind) {
-  case CK_IntegralToBoolean:
-    return Type->isUnsignedIntegerType() ? "0u" : "0";
+  case CK_IntegralToBoolean: {
+    if (Type->isUnsignedIntegerType())
+      return UseUpperCaseLiteralSuffix ? "0U" : "0u";
```

Comment:
Note: You could just use StringRef::upper to avoid using UseUpperCaseLiteralSuffix multiple times.